clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerInterceptors.java

```diff
@@ -76,7 +78,8 @@ public ProducerInterceptors(List<ProducerInterceptor<K, V>> interceptors) {
      *
      * This method does not throw exceptions. Exceptions thrown by any of interceptor methods are caught and ignored.
      *
-     * @param metadata The metadata for the record that was sent (i.e. the partition and offset). Null if an error occurred.
+     * @param metadata The metadata for the record that was sent (i.e. the partition and offset).
+     *                 If an error occurred, metadata will only contain valid topic and maybe partition.
```

Comment:
Got it. However, we probably want to make the api in interceptor consistent with producer Callback. Similar things can happen in the producer callback where we don't have the full metadata, but just the topic/partition. Currently, we just pass a null RecordMetadata. It may be useful to preserve the available metadata there. It's just that changing the exception type in Callback is an incompatible api change.